src/java/fr/paris/lutece/portal/service/securityheader/SecurityHeaderService.java

```diff
@@ -0,0 +1,300 @@
+	private static final String PROPERTY_LABEL_SECURITY_HEADER_PAGE_CATEGORY_LOGOUT_FO = "portal.securityheader.labelLogoutFOPageCategory";
+	private static final String PROPERTY_LABEL_SECURITY_HEADER_PAGE_CATEGORY_ADMIN_BO = "portal.securityheader.labelAdminBOCategory";
+	private static final String PROPERTY_LABEL_SECURITY_HEADER_PAGE_CATEGORY_ADMIN_FO =  "portal.securityheader.labelAdminFOCategory";
+	
+	private Logger _logger = LogManager.getLogger( "lutece.securityHeader" );
+	
+	/**
+	 * Returns all the security headers that match specified name, type and page category (if not null for the latter).
+	 * If page category is null (REST api case), this criterion is ignored.
+	 * 
+	 * @param strName
+	 *           The name of the security header
+	 * @param strType
+	 *           The type of the security header
+	 * @param pageCategory
+	 *           The page category of the security header
+	 * @return list od security headers matching the criteria
+	 */
+	public List<SecurityHeader> find( String strName, String strType, String pageCategory )
+	{
+		ArrayList<SecurityHeader> securityHeadersResultList = new ArrayList<SecurityHeader>( );
+		
+		for( SecurityHeader securityHeader : findAll( ) )
+		{
+			if( securityHeader.getName( ).equals( strName ) && securityHeader.getType( ).equals( strType ) )
+			{
+				if( pageCategory == null || securityHeader.getPageCategory( ).equals( pageCategory ) )
+				{
+					securityHeadersResultList.add( securityHeader );
+				}				
+			}
+		}
+		
+		return securityHeadersResultList;
+	}
+	
+	/**
+	 * Returns all security headers from the specified type and the specified category (if not null)
+	 * that are active (attribute is_enable = true).
+	 * 
+	 * @param strType
+	 * @param strPageCategory
+	 * @return collection on security headers
+	 */
+	public Collection<SecurityHeader> findActive( String strType, String strPageCategory )
+	{
+		ArrayList<SecurityHeader> securityHeadersResultList = new ArrayList<SecurityHeader>( );
+		
+		//
+		for( SecurityHeader securityHeader : findAll( ) )
+		{
+			if( securityHeader.isActive( ) && securityHeader.getType( ).equals( strType ) )
+            {   
+				// Security header page type
+				if( securityHeader.getType( ).equals( SecurityHeaderType.PAGE.getCode( ) ) )
+				{
+					if( strPageCategory.equals( securityHeader.getPageCategory( ) ) )
+					{
+						securityHeadersResultList.add( securityHeader );
+					}
+				} 
+				else // Security header REST api type
+				{
+					securityHeadersResultList.add( securityHeader );
+				}
+            }
+		}
+		
+		return securityHeadersResultList;
+	}
+	
+	/**
+	 * Returns a collection of all security headers.
+	 * Security headers are fetched from database (if data are stale) or from security headers map in memory (if the data in map are still up to date).
+	 * 
+	 * @return collection of security headers
+	 */
+	public Collection<SecurityHeader> findAll( )
```

Comment:
I hesitated a lot between Collection and List types. I've chosen Collection because, as you say, "findAll" method will be called very often and just by calling "values" method on security headers map, I get the collection of headers that filters classes directly (No need of sort or something like that). If you want, I can change for List instead of Collection. Moreover, with the modifications you asked for, findAll method isn't called much anymore